src/weights.jl

```diff
@@ -1,9 +1,9 @@
 
 ###### Weight vector #####
 
-immutable WeightVec{W,Vec<:RealVector}
-    values::Vec
-    sum::W
+immutable WeightVec{T, V<:RealVector, S} <: RealVector{T}
```

Comment:
@nalimilan I've updated the parameterization to match your suggestion, but it needed version check depending on whether the julia version supports triangular dispatch (implemented in [this PR](https://github.com/JuliaLang/julia/pull/18457)).